src/test/fuzz/txrequest.cpp

```diff
@@ -0,0 +1,363 @@
+            // Mark any expired REQUESTED announcements as COMPLETED.
+            for (int peer2 = 0; peer2 < MAX_PEERS; ++peer2) {
+                Announcement& ann2 = m_announcements[txhash][peer2];
+                if (ann2.m_state == State::REQUESTED && ann2.m_time <= m_now) {
+                    ann2.m_state = State::COMPLETED;
+                    break;
+                }
+            }
+            // And delete txids with only COMPLETED announcements left.
+            Cleanup(txhash);
+            // CANDIDATEs for which this announcement has the highest priority get returned.
+            const Announcement& ann = m_announcements[txhash][peer];
+            if (ann.m_state == State::CANDIDATE && GetSelected(txhash) == peer) {
+                result.emplace_back(ann.m_sequence, txhash, ann.m_is_wtxid);
+            }
+        }
+        // Sort the results by sequence number.
+        std::sort(result.begin(), result.end());
+
+        // Compare with TxRequestTracker's implementation.
+        const auto actual = m_tracker.GetRequestable(peer, m_now);
+
+        m_tracker.PostGetRequestableSanityCheck(m_now);
+        assert(result.size() == actual.size());
+        for (size_t pos = 0; pos < actual.size(); ++pos) {
+            assert(TXHASHES[std::get<1>(result[pos])] == actual[pos].GetHash());
+            assert(std::get<2>(result[pos]) == actual[pos].IsWtxid());
+        }
+    }
+
+    void Check()
+    {
+        // Compare CountTracked and CountLoad with naive structure.
+        size_t total = 0;
+        for (int peer = 0; peer < MAX_PEERS; ++peer) {
+            size_t tracked = 0;
+            size_t inflight = 0;
+            size_t candidates = 0;
+            for (int txhash = 0; txhash < MAX_TXHASHES; ++txhash) {
+                tracked += m_announcements[txhash][peer].m_state != State::NOTHING;
+                inflight += m_announcements[txhash][peer].m_state == State::REQUESTED;
+                candidates += m_announcements[txhash][peer].m_state == State::CANDIDATE;
+            }
+            assert(m_tracker.Count(peer) == tracked);
+            assert(m_tracker.CountInFlight(peer) == inflight);
+            assert(m_tracker.CountCandidates(peer) == candidates);
+            total += tracked;
+        }
+        // Compare Size.
+        assert(m_tracker.Size() == total);
+
+        // Invoke internal consistency check of TxRequestTracker object.
+        m_tracker.SanityCheck();
+    }
+};
+} // namespace
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    // Tester object (which encapsulates a TxRequestTracker).
+    Tester tester;
+
+    // Decode the input as a sequence of instructions with parameters
+    auto it = buffer.begin();
+    while (it != buffer.end()) {
+        int cmd = *(it++) % 11;
+        int peer, txidnum, delaynum;
+        switch (cmd) {
+        case 0: // Make time jump to the next event (m_time of PENDING or REQUESTED)
```

Comment:
8d0dd46b4fcdf4c664747e638eda046f8d51079c

`PENDING` ?

Reply:
`CANDIDATE`

Reply:
Done. Indeed, `CANDIDATE`.

Reply:
You replaced the wrong word. It should be CANDIDATE or REQUESTED.

Reply:
Details.

Fixed.